crates/client/rpc/src/lib.rs

```diff
@@ -440,8 +440,8 @@ where
     fn chain_id(&self) -> RpcResult<Felt> {
         let best_block_hash = self.client.info().best_hash;
 
-        let chain_id = self.overrides.for_block_hash(self.client.as_ref(), best_block_hash).chain_id(best_block_hash);
-        Ok(Felt(chain_id.ok_or(StarknetRpcApiError::InternalServerError)?.into()))
+        let chain_id = self.client.runtime_api().chain_id(best_block_hash);
+        Ok(Felt(chain_id.map_err(|_| StarknetRpcApiError::InternalServerError)?.into()))
```

Comment:
i am not sure how to get it, there is `Pallet::chain_id` but that requires a generic T which implements config. there is alot of macro magic going in rumtime which i don't know the inner working of. do you know of a way to make this work?

Reply:
you get it from there

```rust
parameter_types! {
    pub const UnsignedPriority: u64 = 1 << 20;
    pub const TransactionLongevity: u64 = u64::MAX;
    pub const InvokeTxMaxNSteps: u32 = 1_000_000;
    pub const ValidateMaxNSteps: u32 = 1_000_000;
    pub const ProtocolVersion: u8 = 0;
    // Starknet testnet SN_GOERLI
    // Need to use `from_mont` because this needs to be a constant function call
    pub const ChainId: Felt252Wrapper = Felt252Wrapper(FieldElement::from_mont(
        [
            3753493103916128178,
            18446744073709548950,
            18446744073709551615,
            398700013197595345,
        ]
    ));
}
```

`ChainId` is your type, call `get()` on it and you will get it's value

Reply:
ah because that module was private i didn't look into it. had to make it public to make it work.